### PR TITLE
get linux sources from git

### PR DESCRIPTION
**Pull request by razr**
- get Linux sources from git to be able to use any available tag instead of retrieving it via ```apt source```, where old tags were not accessible anymore
   - modify ```rpi4-kernel-build``` action
   - modify Dockerfile
 - add two new actions; 
   - ```Build stable``` - builds `1058` raspi version by default, it is possible to modify it to use any other version before the workflow starts  
   - ```Build latest``` - builds the lastes available raspi version
 - ```rpi4-kernel-build``` is converted to the callable workflow, that is used by both actions above
 -  add additional input parameters to be able to use ```rpi4-kernel-build``` with new kernel versions e.g, 5.15
 - add ```CONFIG_DEBUG_INFO=n```, it removes ```dbg``` kernel Debian package from the build and decreases the size of generated artifacts
 - update documentation
 - [two builds were produced](https://github.com/razr/linux-real-time-kernel-builder/actions) and tested on raspberry pi4

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -84,68 +72,81 @@ RUN apt-get update && apt-get install -q -y sudo \
...
+    && wget http://ports.ubuntu.com/pool/main/l/linux-raspi/linux-buildinfo-${KERNEL_VERSION}-`cat /home/user/uname_r | cut -d '-' -f 2`-raspi_${KERNEL_VERSION}-`cat /home/user/linux_build/tag | cut -d '-' -f 4`_${ARCH}.deb \
+    && dpkg -X *.deb /home/user/
 
 # install lttng dependencies
 RUN sudo apt-get update \
-  && sudo apt-get install -y libuuid1 libpopt0 liburcu6 libxml2 numactl
+    && sudo apt-get install -y libuuid1 libpopt0 liburcu6 libxml2 numactl
 
-COPY ./getpatch.sh /getpatch.sh
+COPY ./getpatch.sh /home/user/.
```
Comment (LanderU) [inline Dockerfile:106]:
```suggestion
COPY ./getpatch.sh /home/user/
```

File: Dockerfile [hunk 2]
```diff
@@ -84,68 +72,81 @@ RUN apt-get update && apt-get install -q -y sudo \
...
+
+WORKDIR /home/user
 
 # run lttng built-in script to configure RT kernel
 RUN set -x \
-  && export KERNEL_DIR=`ls -d */` \
-  && cd $HOME \
-  && cd `ls -d lttng-*/` \
-  && ./scripts/built-in.sh ${HOME}/linux_build/${KERNEL_DIR}
+    cd /home/user \
```
Comment (LanderU) [inline Dockerfile:129]: Do we need this if we set the workdir before?

File: Dockerfile [hunk 3]
```diff
@@ -84,68 +72,81 @@ RUN apt-get update && apt-get install -q -y sudo \
...
+    && cd `ls -d lttng-modules-*` \
+    && ./scripts/built-in.sh ${HOME}/linux_build/${KERNEL_DIR}
 
-# patch `raspi` kernel, do not fail if some patches are skipped
-RUN cd `ls -d */` \
+# patch kernel, do not fail if some patches are skipped
+RUN cd /home/user/linux_build/${KERNEL_DIR} \
```
Comment (LanderU) [inline Dockerfile:134]: Same question here.

File: Dockerfile [hunk 4]
```diff
@@ -84,68 +72,81 @@ RUN apt-get update && apt-get install -q -y sudo \
...
     && OUT="$(patch -p1 --forward < ../patch-`cat $HOME/rt_patch`.patch)" || echo "${OUT}" | grep "Skipping patch" -q || (echo "$OUT" && false);
 
 # setup build environment
-RUN cd `ls -d */` \
+RUN cd /home/user/linux_build/${KERNEL_DIR} \
```
Comment (LanderU) [inline Dockerfile:138]: And here.

File: Dockerfile (hunk 2 above)
Comment (razr) [reply Dockerfile:129]: to execute this script you need to go inside the `lttng` directory. With your change it just `cd /home/user`
Am I missing something?

**Comment by shuhaowu** [comment]
Is this the PR that will resolve the issue with the current GitHub builder where the version number tagged on github doesn't match the debian package built? Once that is resolved, we can probably release the Raspberry Pi 4 image.

I'm also going to submit a 10 minute talk at ROSCon, so definitely need to get it done by then.

**Comment by razr** [comment]
@shuhaowu yes, this is the PR, will merge it this week

File: Dockerfile (hunk 2 above)
Comment (razr) [reply Dockerfile]: I have removed WORKDIR and pushed an update

File: Dockerfile (hunk 3 above)
Comment (razr) [reply Dockerfile:131]: see above :)

File: Dockerfile (hunk 4 above)
Comment (razr) [reply Dockerfile:135]: see above :)

**Comment by LanderU** [comment]
@razr, thank you! I'll do the final review tomorrow and we can merge it. Thanks a lot!

**Review by LanderU — APPROVED** [review]
Thanks a lot!

**Comment by LanderU** [comment]
@carlossvg, can you approve this as well?

File: README.md [hunk 5]
```diff
@@ -4,7 +4,7 @@
 
 ## Introduction
 
-This README describes necessary steps to build and install ```RT_PREEMPT``` Linux kernel for the Raspberry Pi4 board. RT Kernel is a part of the ROS2 real-time system setup. Raspberry Pi4 is a reference board used by the ROS 2 real-time community for the development. RT Kernel is configured as described in [Kernel configuration section](#kernel-configuration). Kernel is built automatically by the Github action, and the artifacts are located under the [```RPI4 RT Kernel build```](https://github.com/ros-realtime/linux-real-time-kernel-builder/actions/workflows/rpi4-kernel-build.yml). Please follow [installation instructions](#deploy-new-kernel-on-raspberry-pi4) to deploy a new kernel to the RPI4 board.
```
Comment (carlossvg) [inline README.md:7]: @razr Shoudn't this be https://github.com/ros-realtime/linux-real-time-kernel-builder/actions/workflows/rpi4-kernel-build.yml?

Comment (razr) [reply README.md:7]: `rpi4-kernel-build.yml` is a callable workflow (like a function) now. There are two other workflows that call it: `build-stable`, and `build-latest`. `build-stable` will build the kernel of our choice (it is possible to pass parameters before the build starts). and `build-latest` will always build the latest kernel. That's why I have replaced the reference to the `rpi4-kernel-build` with  `build-stable`.

**Review by carlossvg — APPROVED** [review]

**Pull request merged**